api/src/main/java/com/example/api/service/activity/result/ranking/RankingService.java

```diff
@@ -82,6 +97,55 @@ public List<RankingResponse> getSearchedRanking(String search) {
         return rankingList;
     }
 
+    public List<RankingResponse> getActivityRanking(Long activityID) throws WrongUserTypeException, EntityNotFoundException {
+        String professorEmail = authService.getAuthentication().getName();
+        User professor = userRepo.findUserByEmail(professorEmail);
+        userValidator.validateProfessorAccount(professor, professorEmail);
+
+        List<RankingResponse> rankingList =  userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
+                        .stream()
+                        .map(user -> studentAndPointsToRankingEntry(user, getStudentPointsForActivity(activityID, user)))
+                        .toList();
+        addPositionToRankingList(rankingList);
+        return rankingList;
+    }
+
+    public List<RankingResponse> getActivityRankingSearch(Long activityID, String search) throws WrongUserTypeException, EntityNotFoundException {
+        String searchLower = search.toLowerCase();
+        List<RankingResponse> rankingList = getActivityRanking(activityID)
+                .stream()
+                .filter(rankingResponse ->
+                    rankingResponse.getFirstName().toLowerCase().equals(searchLower) ||
+                            rankingResponse.getLastName().toLowerCase().equals(searchLower) ||
+                            rankingResponse.getHeroType().getPolishTypeName().toLowerCase().equals(searchLower) ||
+                            rankingResponse.getGroupName().toLowerCase().equals(searchLower)
+                )
+                .sorted(Comparator.comparingDouble(RankingResponse::getPoints).reversed())
+                .toList();
+        addPositionToRankingList(rankingList);
+        return rankingList;
+
+    }
+
+    private Double getStudentPointsForActivity(Long activityID, User user) {
+        GraphTask graphTask = graphTaskRepo.findGraphTaskById(activityID);
+        if (graphTask != null) {
+            GraphTaskResult result = graphTaskResultRepo.findGraphTaskResultByGraphTaskAndUser(graphTask, user);
+            return result != null ? result.getPointsReceived() : 0D;
+        }
+        FileTask fileTask = fileTaskRepo.findFileTaskById(activityID);
+        if (fileTask != null) {
+            FileTaskResult result = fileTaskResultRepo.findFileTaskResultByFileTaskAndUser(fileTask, user);
+            return result != null ? result.getPointsReceived() : 0D;
+        }
+        Survey survey = surveyRepo.findSurveyById(activityID);
+        if (survey != null) {
+            SurveyResult result = surveyResultRepo.findSurveyResultBySurveyAndUser(survey, user);
+            return result != null ? result.getPointsReceived() : 0D;
+        }
+        return null;
+    }
```

Comment:
Uświadomiłem sobie w tym momencie, że samo activityId nie wystarczy. Musimy podać jeszcze activityType, żeby wiedzieć po których aktywnościach szukać. Jak podam gołe 1 z frontu to nie wiemy o które punkty chodzi (czy zadanie bojowe czy ekspedycja czy sondaż). 
@ajablonsk1 Ty kiedyś mi tłumaczyłeś, ten wątek. Wyjaśnij teraz Bartkowi, bo Bartek ma wątpliwości co do unikalności id. Podrzucam też screen ze swaggera (task/activities):
![image](https://user-images.githubusercontent.com/63460537/187021265-375f5481-31c3-4fdc-a098-4de69ef91fc3.png)

Reply:
teraz activityID powinno być unikatowe dla wszystkich rodzajów aktywności